app/styles/ui/_side-by-side-diff.scss

```diff
@@ -138,6 +138,7 @@
     width: 20px;
     left: calc(50% - 10px);
     top: 0;
+    z-index: 1;
```

Comment:
We'd like to avoid setting the `z-index` because it could end up leading to complex situations where we have to assign random z-index number to different elements to make everything work as expected.

Instead, I've looked into this particular bug, and it seems we can fix it by removing this line…
```suggestion
```
…and reordering the hunk handle when it's rendered. You can apply these changes to fix it in all cases:
```diff
diff --git a/app/src/ui/diff/side-by-side-diff-row.tsx b/app/src/ui/diff/side-by-side-diff-row.tsx
index f1decce05..bd94942b4 100644
--- a/app/src/ui/diff/side-by-side-diff-row.tsx
+++ b/app/src/ui/diff/side-by-side-diff-row.tsx
@@ -204,12 +204,12 @@ export class SideBySideDiffRow extends React.Component<
               {this.renderContentFromString('')}
               {this.renderWhitespaceHintPopover(DiffColumn.Before)}
             </div>
-            {this.renderHunkHandle()}
             <div className="after">
               {this.renderLineNumber(lineNumber, isSelected)}
               {this.renderContent(row.data)}
               {this.renderWhitespaceHintPopover(DiffColumn.After)}
             </div>
+            {this.renderHunkHandle()}
           </div>
         )
       }
@@ -241,12 +241,12 @@ export class SideBySideDiffRow extends React.Component<
               {this.renderContent(row.data)}
               {this.renderWhitespaceHintPopover(DiffColumn.Before)}
             </div>
-            {this.renderHunkHandle()}
             <div className="after">
               {this.renderLineNumber()}
               {this.renderContentFromString('')}
               {this.renderWhitespaceHintPopover(DiffColumn.After)}
             </div>
+            {this.renderHunkHandle()}
           </div>
         )
       }
@@ -259,12 +259,12 @@ export class SideBySideDiffRow extends React.Component<
               {this.renderContent(before)}
               {this.renderWhitespaceHintPopover(DiffColumn.Before)}
             </div>
-            {this.renderHunkHandle()}
             <div className="after" onMouseEnter={this.onMouseEnterLineNumber}>
               {this.renderLineNumber(after.lineNumber, after.isSelected)}
               {this.renderContent(after)}
               {this.renderWhitespaceHintPopover(DiffColumn.After)}
             </div>
+            {this.renderHunkHandle()}
           </div>
         )
       }
```

One of the changes affect added lines, which already worked… except for this annoying glitch:
![Kapture 2021-11-16 at 13 24 56](https://user-images.githubusercontent.com/1083228/141984955-d10e90b5-5b60-4db0-9a2d-f9dfb6aa051d.gif)

With the diff above, it will keep the same cursor over the area of the whole hunk handle 😄

Reply:
Thanks, that was the other possible solution that I was thinking. I've applied the suggested changes 🙂